src/prefetch.cpp

```diff
@@ -0,0 +1,326 @@
+/*
+ * prefetch.cpp
+ *
+ * Copyright (C) 2021 Alibaba Group.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * See the file COPYING included with this distribution for more details.
+ */
```

Comment:
@beef9999 GPL2 is incompatible with Apache License 2.0.
Could you revert this PR, for get Alibaba to relicense the file for Apache License 2.0?

Reply:
This part of code was developed during the period that we were trying to donate alibaba/overlaybd to CNCF containerd. This is a mistake. I will switch to Apache license.

Reply:
Thank you!